superset-frontend/src/addSlice/AddSliceContainer.tsx

```diff
@@ -126,19 +118,47 @@ export default class AddSliceContainer extends React.PureComponent<
     window.location.href = this.exploreUrl();
   }
 
-  changeDatasource(value: string) {
-    this.setState({
-      datasourceValue: value,
-      datasourceId: value.split('__')[0],
-    });
+  changeDatasource(datasource: { label: string; value: string }) {
+    this.setState({ datasource });
   }
 
   changeVisType(visType: string | null) {
     this.setState({ visType });
   }
 
   isBtnDisabled() {
-    return !(this.state.datasourceId && this.state.visType);
+    return !(this.state.datasource?.value && this.state.visType);
+  }
+
+  loadDatasources(search: string, page: number, pageSize: number) {
+    const query = rison.encode({
+      columns: ['id', 'table_name', 'description', 'datasource_type'],
+      filter: search,
+      page,
+      page_size: pageSize,
+    });
+    return SupersetClient.get({
+      endpoint: `/api/v1/dataset?q=${query}`,
+    }).then(response => {
+      const list = response.json.result.map(
+        (item: {
+          id: number;
+          table_name: string;
+          description: string;
+          datasource_type: string;
+        }) => ({
+          value: `${item.id}__${item.datasource_type}`,
+          label: item.table_name,
+          title: `${item.table_name}\n\n${
```

Comment:
Can we replace `\n\n` with a span with some padding, so that we have more control over the spacings and can use grid unit from theme?